taichi/ir/ir.h

```diff
@@ -111,6 +111,8 @@ void demote_atomics(IRNode *root);
 void reverse_segments(IRNode *root);  // for autograd
 std::unique_ptr<ScratchPads> initialize_scratch_pad(StructForStmt *root);
 std::unordered_set<SNode *> gather_deactivations(IRNode *root);
+std::vector<Stmt *> gather_statements(IRNode *root,
+                                      const std::function<bool(Stmt *)> &test);
```

Comment:
Just wondering, why does this function (with `detect_fors_with_break` and `gather_deactivations`) appear in `namespace irpass`, but not `namespace analysis` as `value_diff` do? The cpp files are all in `analysis/`.